sdk/provisioning/Azure.Provisioning.Storage/tests/BasicStorageTests.cs

```diff
@@ -0,0 +1,389 @@
+            })
+        .Compare(
+            """
+            resource storage 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+                name: take('storage${uniqueString(resourceGroup().id)}', 24)
+                kind: 'StorageV2'
+                location: resourceGroup().location
+                sku: {
+                    name: 'Standard_LRS'
+                }
+                properties: {
+                    allowBlobPublicAccess: false
+                    isHnsEnabled: true
+                }
+            }
+
+            resource blobs 'Microsoft.Storage/storageAccounts/blobServices@2023-01-01' = {
+                name: 'default'
+                parent: storage
+            }
+
+            output blobs_endpoint string = storage.properties.primaryEndpoints.blob
+            """)
+        .Lint()
+        .ValidateAndDeployAsync();
+    }
+
+    [RecordedTest]
+    [Description("https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.storage/storage-account-create/main.bicep")]
+    public async Task CreateStandardStorageAccount()
+    {
+        await using Trycep test = CreateBicepTest();
+        await test.Define(
+            ctx =>
+            {
+                BicepParameter storageAccountType = BicepParameter.Create<string>(nameof(storageAccountType), StorageSkuName.StandardLrs);
+                storageAccountType.Description = "Storage Account type";
+
+                BicepParameter location = BicepParameter.Create<string>(nameof(location), BicepFunction.GetResourceGroup().Location);
+                location.Description = "The storage account location.";
+
+                StorageAccount sa =
+                    new(nameof(sa))
+                    {
+                        Location = location,
+                        Sku = new StorageSku { Name = storageAccountType },
+                        Kind = StorageKind.StorageV2
+                    };
+
+                BicepOutput.Create<string>("storageAccountName", sa.Name);
+                BicepOutput.Create<string>("storageAccountId", sa.Id);
+            })
+        .Compare(
+            """
+            @description('Storage Account type')
+            param storageAccountType string = 'Standard_LRS'
+
+            @description('The storage account location.')
+            param location string = resourceGroup().location
+
+            resource sa 'Microsoft.Storage/storageAccounts@2023-01-01' = {
+                name: take('sa${uniqueString(resourceGroup().id)}', 24)
+                kind: 'StorageV2'
+                location: location
+                sku: {
+                    name: storageAccountType
+                }
+            }
+
+            output storageAccountName string = sa.name
+
+            output storageAccountId string = sa.id
+            """)
+        .Lint()
+        .ValidateAndDeployAsync();
```

Comment:
Hello, Storage.  For all these tests, the lambda is what customers would write and the giant string is the generated Bicep.